connectors/src/connectors/google_drive/index.ts

```diff
@@ -304,43 +308,221 @@ export async function retrieveGoogleDriveConnectorPermissions(
     logger.error({ connectorId }, "Connector not found");
     return new Err(new Error("Connector not found"));
   }
+  const authCredentials = await getAuthObject(c.connectionId);
+  if (filterPermission === "read") {
+    if (parentInternalId === null) {
+      // Return the list of folders explicitly selected by the user.
+      const folders = await GoogleDriveFolders.findAll({
+        where: {
+          connectorId: connectorId,
+        },
+      });
 
-  const folders = await GoogleDriveFolders.findAll({
-    where: {
-      connectorId: connectorId,
-    },
-  });
+      const driveClient = await getDriveClient(authCredentials);
 
-  const driveClient = await getDriveClient(c.connectionId);
+      const resources: ConnectorResource[] = await Promise.all(
+        folders.map((f) => {
+          return (async () => {
```

Comment:
why do you need to return a self-executing async arrow function here ? Can't you make the `.map()` callback an async fn directly ?